src/components/DiffView/index.spec.tsx

```diff
@@ -276,6 +277,39 @@ describe(__filename, () => {
     });
   });
 
+  it('does not tokenize the hunks to add syntax highlighting if the diff has been trimmed', () => {
+    const _codeCanBeHighlighted = jest.fn().mockReturnValue(true);
+    const _codeShouldBeTrimmed = jest.fn().mockReturnValue(true);
+    const _tokenize = jest.fn();
+
+    renderWithLinterProvider({
+      _codeCanBeHighlighted,
+      _codeShouldBeTrimmed,
+      _tokenize,
+      // A minified file will be trimmed by default.
+      isMinified: true,
+    });
+
+    expect(_tokenize).not.toHaveBeenCalled();
+  });
+
+  it('resets tokens to undefined if they seem invalid', () => {
+    const _codeCanBeHighlighted = jest.fn().mockReturnValue(true);
+    const _codeShouldBeTrimmed = jest.fn().mockReturnValue(false);
+    const _tokenize = jest
+      .fn()
+      .mockReturnValue(createFakeTokens({ invalid: true }));
```

Comment:
Could we use the `tokenize` function here, instead of a fake object? The code in the component is not very future-proof, could easily break and we don't have coverage around the returned object of `tokenize` when it has failed.